### PR TITLE
[NGraph] CTCLoss reference implementation.

### PR DESCRIPTION
**Pull request by nshchego**

### CODE REVIEW

File: ngraph/src/ngraph/runtime/reference/ctc_loss.hpp [hunk 1]
```diff
@@ -0,0 +1,137 @@
+// Copyright (C) 2020 Intel Corporation
```
Comment (ilyachur) [inline ngraph/src/ngraph/runtime/reference/ctc_loss.hpp:1]: If we don't use this implementation in the nGraph library, please move it to test folder.

File: ngraph/src/ngraph/runtime/reference/ctc_loss.hpp [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+namespace ngraph
+{
+namespace runtime
+{
+namespace reference
+{
+
+template <typename T, typename U>
+void CTCLoss(const T* logits,
```
Comment (ilyachur) [inline ngraph/src/ngraph/runtime/reference/ctc_loss.hpp:17]: Can you add tests for this implementation?

File: ngraph/src/ngraph/runtime/reference/ctc_loss.hpp (hunk 1 above)
Comment (nshchego) [reply ngraph/src/ngraph/runtime/reference/ctc_loss.hpp]: FIxed

**Comment by nshchego** [comment]
> Please fix a code style

Fixed

File: ngraph/src/ngraph/runtime/reference/ctc_loss.hpp (hunk 2 above)
Comment (nshchego) [reply ngraph/src/ngraph/runtime/reference/ctc_loss.hpp]: It's supposed, that plugins will test that. CPU for example https://github.com/openvinotoolkit/openvino/pull/1482

File: ngraph/test/runtime/reference/ctc_loss.hpp [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+                    std::function<void(size_t targetIdx, size_t start, size_t end)> findPaths =
+                        [&](size_t targetIdx, size_t start, size_t end) {
+                            if (end > actualLogitLen)
+                            {
+                                T ex = 0;
+                                T denom = 1;
+                                for (size_t t = 0; t < actualLogitLen; t++)
+                                {
+                                    ex += logits[BTC + classesNum * t + pathS[t]];
+                                    denom *= kExp[t];
```
Comment (rkazants) [inline ngraph/test/runtime/reference/ctc_loss.hpp:118]: Use log(softmax(logit)) to compute log(prob(seq)) as a sum of log(softmax(...)) of logits. After that please sum up log(prob)) for sequences using formula ln(a + b) = ln(a) + ln(1 + exp(ln(b) - ln(a))): https://github.com/githubharald/CTCWordBeamSearch/issues/23

Comment (nshchego) [reply ngraph/test/runtime/reference/ctc_loss.hpp]: Fixed

File: ngraph/test/runtime/reference/ctc_loss.hpp [hunk 4]
```diff
@@ -0,0 +1,180 @@
+// Copyright (C) 2020 Intel Corporation
```
[the remark this answers is not in the document]
Comment (nshchego) [reply ngraph/test/runtime/reference/ctc_loss.hpp]: Fixed
